app/src/main/java/com/github/polybooks/core/database/interfaces/BookDatabase.kt

```diff
@@ -66,8 +66,28 @@ interface BookQuery : Query<Book>, Serializable {
      * */
     fun withOrdering(ordering : BookOrdering) : BookQuery
 
+    /**
+     * Get Settings of the BookQuery at current state
+     **/
+    fun getSettings() : BookSettings
+
+    /**
+     * Reset this query using the given settings
+     */
+    fun fromSettings(settings : BookSettings) : BookQuery
+
 }
 
+/**
+ * The settings contains the values for all the possible parameters of the Query
+ * In contrary to Query, it is independent to the state of the database.
+ */
+data class BookSettings(
+        val ordering: BookOrdering,
+        val isbn13: String?,
```

Comment:
Probably change the name "isbn13" to "isbn" to match what we have in the latests two PR : #108 and #109 (they changed isbn13 to isbn)

```suggestion
        val isbn: String?,
```